### PR TITLE
Add Image Grid toggle for run parameters

### LINKED ISSUE #23512: [FR] [UI] Image Grid chart run-params header overflows when runs have many params (regression in 3.12.0)
> [!WARNING]
> Before submitting a PR, please make sure that:
> - A maintainer has triaged this issue and applied the `ready` label
> - This issue has no assignee
> - No duplicate PR exists
>
> PRs not meeting these requirements may be automatically closed.

### Issues Policy acknowledgement
- [x] I have read and agree to submit bug reports in accordance with the issues policy

### Where did you encounter this bug?
Local machine

### MLflow version
- Client: 3.12.0
- Tracking server: 3.12.0

### System information
- OS: macOS (Darwin 25.3.0)
- Browser: Chrome
- Python: 3.x

### Describe the problem
PR #22118 (shipped in 3.12.0) added a header above each image in the Image Grid chart that lists the run's parameters (intent: *"show run params (rank, optimizer, lr) in column headers so you can tell configs apart"* — see linked FR #22117). This works well for the curated 2–4 param case the FR describes, but degrades badly when the run has many params.

In my case the run is an object-detection training job whose hyperparameter dict contains an 80-class name mapping. MLflow's normal `log_hyperparams` flattening turns this into ~100 individual params (`data/names/0=person`, `data/names/1=bicycle`, ...). The Image Grid header then renders the entire flattened list concatenated as `key=value, key=value, ...` above every image cell, taking up far more vertical space than the image itself and effectively burying the visualization.

Pinning `mlflow<3.12` (e.g. 3.11.1) removes the header and the chart is usable again, confirming this is the regression point.

### Steps to reproduce the bug
1. Start MLflow 3.12.0+ tracking server.
2. Run the script in the next section.
3. Open the run in the MLflow UI → Charts → Image Grid for `val_images/predictions`.
4. Observe that each image cell has a ~100-line header listing every flattened param, dwarfing the image.

### Code to generate data required to reproduce the bug
```python
import mlflow
from PIL import Image

mlflow.set_tracking_uri("http://localhost:5000")
mlflow.set_experiment("image-header-repro")

with mlflow.start_run():
    # Simulate a flattened many-key hyperparams dict (e.g. a class-name mapping).
    for i in range(80):
        mlflow.log_param(f"data/names/{i}", f"class_{i}")
    # A handful of "real" hyperparams on top.
    for k, v in {"lr": 0.1, "batch_size": 4, "model": "picodet/l-640", "steps": 100}.items():
        mlflow.log_param(k, v)

    img = Image.new("RGB", (640, 480), (128, 128, 128))
    for step in range(5):
        mlflow.log_image(img, key="val_images/predictions", step=step)
```

### Is the console panel in DevTools showing errors relevant to the bug?
N/A

### Does the network panel in DevTools contain failed requests relevant to the bug?
N/A

### Suggested mitigations
- Truncate the header after N params or N characters with a "show more" affordance.
- Allow per-chart selection of which params appear in the header (allow/deny list).
- Add a per-chart toggle to hide the header entirely.

Related: #22117 (parent FR), #22118 (implementation).

### PR DESCRIPTION
**Pull request by jdoughty04**
### Related Issues/PRs

Closes #23512

### What changes are proposed in this pull request?

This adds a per-chart "Show run parameters" toggle to Image Grid chart configuration. The toggle defaults on, so existing charts keep showing run parameters, but users comparing runs with many parameters can turn the header details off for that chart.

The setting is persisted on the image chart config and applied to both single-key and multiple-key image grid layouts. The run name remains visible and tooltip behavior still falls back to the run name when parameters are hidden.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests

Checks run:

- `corepack yarn test --runTestsByPath src/experiment-tracking/components/runs-charts/components/charts/ImageGridPlot.common.test.tsx --runInBand`
- `corepack yarn eslint --ext js,jsx,ts,tsx src/experiment-tracking/components/runs-charts/components/charts/ImageGridPlot.common.test.tsx src/experiment-tracking/components/runs-charts/components/charts/ImageGridPlot.common.tsx src/experiment-tracking/components/runs-charts/components/charts/ImageGridSingleKeyPlot.tsx src/experiment-tracking/components/runs-charts/components/charts/ImageGridMultipleKeyPlot.tsx src/experiment-tracking/components/runs-charts/components/config/RunsChartsConfigureImageChart.tsx src/experiment-tracking/components/runs-charts/runs-charts.types.ts`
- `corepack yarn i18n:check`
- `corepack yarn type-check`
- `corepack yarn prettier --check src/experiment-tracking/components/runs-charts/components/charts/ImageGridPlot.common.test.tsx src/experiment-tracking/components/runs-charts/components/charts/ImageGridPlot.common.tsx src/experiment-tracking/components/runs-charts/components/charts/ImageGridSingleKeyPlot.tsx src/experiment-tracking/components/runs-charts/components/charts/ImageGridMultipleKeyPlot.tsx src/experiment-tracking/components/runs-charts/components/config/RunsChartsConfigureImageChart.tsx src/experiment-tracking/components/runs-charts/runs-charts.types.ts`
- `git diff --check`

### Does this PR require documentation update?

- [x] No.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

### Does this PR require updating the [MLflow Skills](https://github.com/mlflow/skills) repository?

- [x] No.
- [ ] Yes. Please link the corresponding PR or explain how you plan to update it.

#### Is this a user-facing change?

- [ ] No.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Image Grid charts now include a per-chart option to hide run parameters in the run header, making dense image comparisons easier to read when runs have many parameters.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows
- [ ] `area/gateway`: MLflow AI Gateway client APIs, server, and third-party integrations
- [ ] `area/prompts`: MLflow prompt engineering features, prompt templates, and prompt management
- [ ] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality
- [ ] `area/projects`: MLproject format, project running backends
- [x] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Is this PR a critical bugfix or security fix that should go into the next patch release?

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Minor releases are expected to contain larger changes, such as new features and improvements. Non-critical bug fixes and doc updates can be included as well. By default, your PR should target the next minor release.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Patch releases are typically only performed when there has been a major regression or bug in the latest release. For the sake of stability, your PR should not be included in a patch release unless it is a critical fix, or if the risk level of your PR is exceedingly low.

</details>

- [ ] This PR is critical and needs to be in the next patch release
- [x] This PR can wait for the next minor release

### CODE REVIEW

**Comment by Mxbonn** [comment]
Having the same issue here, this PR would solve it.

**Review by joshuawong-db — COMMENTED** [review]
On the runs page
Image

Inside the configuration page
Screenshot 2026-08-06 at 6 22 13 PM

Thanks for the PR!

The feature works. However, when I toggle the button, it updates the images inside the run chart configuration page, but on the runs page, the runs chart is not updated until I refresh the page. 


We'll want to thread this update into that state as well.

**Review by joshuawong-db — APPROVED** [review]

**Comment by joshuawong-db** [comment]
Thank you for the change @jdoughty04, and apologies for the delay in reviewing. As the nit was relatively minor, I directly pushed to your branch so we can get this fix in ASAP.
